ignitionrobotics/ign-gazebo
# Support editing air pressure sensor in the GUI

Signed-off-by: Nate Koenig <nate@openrobotics.org>

# 🎉 New feature

## Summary

Adds support for editing the noise properties of the air pressure sensor in the component inspector.

Requires:
* https://github.com/ignitionrobotics/ign-gazebo/pull/1109
* https://github.com/ignitionrobotics/sdformat/pull/737

I've made some changes to the component inspector in an attempt to make it more modular, these include:
1. We have ability to (mostly) encapsulate an inspector element into a set of three files consisting of a `.cc`, `.hh`, and `.qml`. See #1172 for an example of adding a new inspector element.
2. There is a bit of ugliness where an inspector element has to [callout to the `componentInspector`](https://github.com/ignitionrobotics/ign-gazebo/pull/1171/files#diff-ffe1a0ce87209ac513a5b088feb12af0e52506e74e95c83702a015b1815baba5R130) in order to then call the [c++ function](https://github.com/ignitionrobotics/ign-gazebo/pull/1171/files#diff-37e11db45de8ed455c819cc8d33a7bbcafb06b3f6cd9c865bdc191e6b81baf16R93). Any help here would be appreciated.
3. The header for a element can be customized. By default the header name is based on the QML filename.

## Test it

1. `ign gazebo -v 4 sensors.sdf`
2. Take note of the output from `ign model -m sensors_box -l link -s air_pressure`
3. Select the altimeter sensor in the entity tree.
4. Make some changes to the values in the component inspector. 
5. Rerun `ign model -m sensors_box -l link -s air_pressure` and you should see the changes.

## Checklist

**Note to maintainers**: Remember to use **Squash-Merge**
